src/app/graphql/resolvers.ts

```diff
@@ -1,34 +1,53 @@
 import prisma from '../client/client';
-import { UserInput } from '../interfaces/user';
-import { LoginInput } from '../interfaces/login';
-import isPasswordValid from './password';
+import { UserInput, LoginInput, User } from '../interfaces/';
+import { isPasswordValid, comparePassword } from './password';
 import bcrypt from 'bcrypt';
 import { CustomError } from '../errors/CustomError';
 import jwt from 'jsonwebtoken';
 import 'dotenv/config';
 import { AuthenticationError } from 'apollo-server';
 import ContextType from './context-type';
+import { verifyToken } from '../client/validation';
 const JWT_SECRET = process.env.JWT_SECRET ?? '';
 
-const comparePassword = async (password: string, hashedPassword: string): Promise<boolean> => {
-  const isMatch = await bcrypt.compare(password, hashedPassword);
-  return isMatch;
-};
-
 export const resolvers = {
   Query: {
-    hello: () => 'hello!',
-    users: async () => {
-      const users = await prisma.user.findMany();
+    hello: (): string => 'hello!',
+    users: async (_: unknown, { end }: { end: number }, context: ContextType) => {
+      const { token } = context;
 
+      if (!token) {
+        throw new AuthenticationError('Token is required for this operation!', {
+          http_status: '400',
+          field: 'authorization',
+          reason: 'A valid token must be provided.',
+        });
+      };
+
+      verifyToken(token, JWT_SECRET);
+
+      const users = await prisma.user.findMany();
       if (!users.length) {
         throw new CustomError('404', 'Users not found!', {
           field: 'User',
           reason: 'There are no users.',
         });
-      }
+      };
 
-      return users;
+      const sortUsers = (users: User[]): User[] => {
```

Comment:
não é necessário fazer essa ordenação em javascript, [o prisma já oferece uma resposta ordenada
](https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting#sorting)